### ISSUE TITLE
Error in plot_completeness

### ISSUE THREAD

**Issue by christinawlindberg** [BEAST-Fitting/beast#595]
I think maybe there is an error in the plot_completeness code after the noisemodel values were changed to dictionaries.

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-48-e1fe8aa4dbd4> in <module>
      4                                     param_list=['Av', 'Rv', 'logA', 'f_A', 'M_ini', 'Z'],
      5                                     #, 'distance'],
----> 6                                     compl_filter='F475W',)

~/anaconda3/envs/astroconda/lib/python3.6/site-packages/beast/plotting/plot_completeness.py in plot_completeness(physgrid_list, noise_model_list, output_plot_filename, param_list, compl_filter)
     69         model_compl = noisegrid["completeness"]
     70         # close the file to save memory
---> 71         noisegrid.close()
     72 
     73         # put it all into a table

AttributeError: 'dict' object has no attribute 'close'
```

Getting rid of the line seems to solve the issue but there is still a UserWarning stating that tight_layout is incompatible with some of the Axes, although I'm not sure how relevant it is to fix this.

```---------------------------------------------------------------------------
UserWarning                               Traceback (most recent call last)
<ipython-input-55-f57891a9a25a> in <module>
      4                                     param_list=['Av', 'Rv', 'logA', 'f_A', 'M_ini', 'Z'],
      5                                     #, 'distance'],
----> 6                                     compl_filter='F475W',)

<ipython-input-54-c11825855fd0> in plot_completeness(physgrid_list, noise_model_list, output_plot_filename, param_list, compl_filter)
    205     cbar.set_label("Completeness", fontsize=label_font)
    206     cbar.ax.tick_params(labelsize=tick_font)
--> 207     gs.tight_layout(fig)
    208 
    209     fig.savefig(output_plot_filename)

~/anaconda3/envs/astroconda/lib/python3.6/site-packages/matplotlib/gridspec.py in tight_layout(self, figure, renderer, pad, h_pad, w_pad, rect)
    334             figure.axes, grid_spec=self)
    335         if None in subplotspec_list:
--> 336             cbook._warn_external("This figure includes Axes that are not "
    337                                  "compatible with tight_layout, so results "
    338                                  "might be incorrect.")

~/anaconda3/envs/astroconda/lib/python3.6/site-packages/matplotlib/cbook/__init__.py in _warn_external(message, category)
   2076             break
   2077         frame = frame.f_back
-> 2078     warnings.warn(message, category, stacklevel)
   2079 
   2080 

UserWarning: This figure includes Axes that are not compatible with tight_layout, so results might be incorrect.```

**Comment by karllark** [comment]
Great to see that you found the solution.   This is definitely the answer as #549 fixed the issue of unclosed files, but as you say by returning a dictionary instead of an open file.   Are you planning on putting in a pull request to fix this issue?

**Comment by christinawlindberg** [comment]
I can, I just need to get everything sorted with the M31 example before I mess with anything. I'll do it tomorrow.

**Comment by karllark** [comment]
Most excellent!

**Comment by karllark** [comment]
Addressed by PR #597.
